ably/types/options.py

```diff
@@ -30,8 +30,10 @@ def __init__(self, client_id=None, log_level=0, tls=True, rest_host=None,
             from ably import api_version
             idempotent_rest_publishing = api_version >= '1.2'
 
+        if environment is None:
+            environment = Defaults.environment
         if realtime_host is None:
-            realtime_host = Defaults.realtime_host
+            realtime_host = f'{environment}-{Defaults.realtime_host}'
```

Comment:
I think it's a bad idea to mutate client options like this. Have a look at the `Options.__get_rest_hosts()` method, I think it would be better to implement a simple `Options.__get_realtime_host` method rather than doing it like this.